### PR TITLE
Rename seq# powered optimistic concurrency control parameters to ifSeqNo/ifPrimaryTerm 

### PR DESCRIPTION
**Pull request by bleskes**
This PR renames the parameters previously introduce to the following:

### URL Parameters
```
PUT twitter/_doc/1?if_seq_no=501&if_primary_term=1
{
    "user" : "kimchy",
    "post_date" : "2009-11-15T14:12:12",
    "message" : "trying out Elasticsearch"
}

DELETE twitter/_doc/1?if_seq_no=501&if_primary_term=1
```

### Bulk API
```
POST _bulk
{ "index" : { "_index" : "test", "_type" : "_doc", "_id" : "1", "if_seq_no": 501, "if_primary_term": 1 } }
{ "field1" : "value1" }
{ "delete" : { "_index" : "test", "_type" : "_doc", "_id" : "2", "if_seq_no": 501, "if_primary_term": 1 } }
```

### Java API
```
IndexRequest.ifSeqNo(long seqNo)
IndexRequest.ifPrimaryTerm(long primaryTerm)
DeleteRequest.ifSeqNo(long seqNo)
DeleteRequest.ifPrimaryTerm(long primaryTerm)
```

Relates #36148
Relates #10708

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/index/IndexRequest.java [hunk 1]
```diff
@@ -486,31 +493,28 @@ public IndexRequest versionType(VersionType versionType) {
...
         }
-        ifSeqNoMatch = seqNo;
-        ifPrimaryTermMatch = term;
+        ifPrimaryTerm = term;
         return this;
     }
 
-    public long ifSeqNoMatch() {
-        return ifSeqNoMatch;
+    public long ifSeqNo() {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: can you add javadocs to the new getters and setters in IndexRequest and DeleteRequest?

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by bleskes** [comment]
run default distro tests

**Comment by bleskes** [comment]
run gradle build  tests 1

**Comment by bleskes** [comment]
run gradle build tests 1

**Review by Mpdreamz — APPROVED** [review]
Rename of a previously unreleased query string parameter is all good when back ported to 6.6.

**Pull request merged**